packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -58,70 +59,160 @@ export interface TcpTimeout {
 }
 
 /**
- * Enum of supported AppMesh protocols
+ * Properties used to define healthchecks.
  */
-export enum Protocol {
-  HTTP = 'http',
-  TCP = 'tcp',
-  HTTP2 = 'http2',
-  GRPC = 'grpc',
-}
-
-/**
- * Properties used to define healthchecks when creating virtual nodes.
- * All values have a default if only specified as {} when creating.
- * If property not set, then no healthchecks will be defined.
- */
-export interface HealthCheck {
+export interface HealthCheckCommonOptions {
   /**
-   * Number of successful attempts before considering the node UP
+   * The number of consecutive successful health checks that must occur before declaring listener healthy.
    *
-   * @default 2
+   * @default 5
    */
   readonly healthyThreshold?: number;
 
   /**
-   * Interval in milliseconds to re-check
+   * The time period in milliseconds between each health check execution.
    *
-   * @default 5 seconds
+   * @default Duration.seconds(30)
```

Comment:
Same question here (why this change?).

Reply:
We can change these back if you prefer -- I just don't know how they were originally chosen.

Reply:
@dfezzie I'm a little wary of changing so many things at the same time in a single PR. Is there any way we can split this PR in 2 then?

1. In the first PR, change the API to use the union-like pattern, while keeping the defaults the same.
2. In the second PR, change the defaults.

Or do you think that's too irritating/too much work?

Reply:
I'm fine with that!

Reply:
I'd like to keep dropping the port from the interface though. I'll revert the defaults though